example/mapboxExampleCamera.js

```diff
@@ -0,0 +1,284 @@
+import * as THREE from 'three';
+
+const utils = {
+	clamp: function ( n, min, max ) {
+
+		return Math.min( max, Math.max( min, n ) );
+
+	},
+	makePerspectiveMatrix: function ( fovy, aspect, near, far ) {
+
+		var out = new THREE.Matrix4();
+		var f = 1.0 / Math.tan( fovy / 2 ),
+			nf = 1 / ( near - far );
+		var newMatrix = [
+			f / aspect,
+			0,
+			0,
+			0,
+			0,
+			f,
+			0,
+			0,
+			0,
+			0,
+			( far + near ) * nf,
+			- 1,
+			0,
+			0,
+			2 * far * near * nf,
+			0,
+		];
+
+		out.elements = newMatrix;
+		return out;
+
+	},
+};
+
+// 512 * 2000
+// 512: TILE_SIZE
+const WORLD_SIZE = 10240000;
+const FOV_ORTHO = ( 0.1 / 180 ) * Math.PI; // Mapbox doesn't accept 0 as FOV
+const FOV = Math.atan( 3 / 4 ); // from Mapbox https://github.com/mapbox/mapbox-gl-js/blob/main/src/geo/transform.js#L93
+const EARTH_RADIUS = 6371008.8; // from Mapbox  https://github.com/mapbox/mapbox-gl-js/blob/0063cbd10a97218fb6a0f64c99bf18609b918f4c/src/geo/lng_lat.js#L11
+const EARTH_CIRCUMFERENCE_EQUATOR = 40075017; // from Mapbox https://github.com/mapbox/mapbox-gl-js/blob/0063cbd10a97218fb6a0f64c99bf18609b918f4c/src/geo/lng_lat.js#L117
+const MapConstants = {
+	WORLD_SIZE: WORLD_SIZE,
+	PROJECTION_WORLD_SIZE: WORLD_SIZE / ( EARTH_RADIUS * Math.PI * 2 ),
+	MERCATOR_A: EARTH_RADIUS,
+	DEG2RAD: Math.PI / 180,
+	RAD2DEG: 180 / Math.PI,
+	EARTH_RADIUS: EARTH_RADIUS,
+	EARTH_CIRCUMFERENCE: 2 * Math.PI * EARTH_RADIUS,
+	EARTH_CIRCUMFERENCE_EQUATOR: EARTH_CIRCUMFERENCE_EQUATOR,
+	FOV_ORTHO: FOV_ORTHO,
+	FOV: FOV,
+	FOV_DEGREES: ( FOV * 180 ) / Math.PI,
+	TILE_SIZE: 512,
+};
+
+const Constants = MapConstants;
+
+function projectToWorld( coords ) {
+
+	// Spherical mercator forward projection, re-scaling to WORLD_SIZE
+	var projected = [
+		- Constants.MERCATOR_A * Constants.DEG2RAD * coords[ 0 ] * Constants.PROJECTION_WORLD_SIZE,
+		- Constants.MERCATOR_A * Math.log( Math.tan( Math.PI * 0.25 + 0.5 * Constants.DEG2RAD * coords[ 1 ] ) ) * Constants.PROJECTION_WORLD_SIZE,
+	];
+
+	// z dimension, defaulting to 0 if not provided
+	if ( ! coords[ 2 ] ) projected.push( 0 );
+	else {
+
+		var pixelsPerMeter = projectedUnitsPerMeter( coords[ 1 ] );
+		projected.push( coords[ 2 ] * pixelsPerMeter );
+
+	}
+
+	var result = new THREE.Vector3( projected[ 0 ], projected[ 1 ], projected[ 2 ] );
+
+	return result;
+
+}
+
+function projectedUnitsPerMeter( latitude ) {
+
+	return Math.abs(
+		Constants.WORLD_SIZE / Math.cos( Constants.DEG2RAD * latitude ) / Constants.EARTH_CIRCUMFERENCE );
+
+}
+
+class CameraSync {
+
+	constructor( map, camera, world ) {
+
+		this.map = map;
+		this.camera = camera;
+		this.active = true;
+		// We're in charge of the camera now!
+		this.camera.matrixAutoUpdate = false;
+		// Postion and configure the world group so we can scale it appropriately when the camera zooms
+		this.world = world || new THREE.Group();
+		this.world.position.x = this.world.position.y =
+    MapConstants.WORLD_SIZE / 2;
```

Comment:
The rationale for some of these changes should be documented, I think. What's the reasoning for setting the position here? We disable matrix auto update and set the matrix explicitly later so this will never really be used, right?